web/handlers.go

```diff
@@ -79,6 +80,7 @@ type Handler struct {
 }
 
 func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	newWriter := BuildResponseWriter(w)
```

Comment:
Is there a need to create a new variable ? Since we are wrapping it, it maybe better to just overwrite w with this and then reuse w throughout. And we can avoid this code churn.

Also, there is no need to expose `BuildResponseWriter`. Idiomatically, it should be `newWrappedWriter(w)`.

Reply:
I don't like to modify the current object to avoid side-effects or if we need the original one. I'd rather prefer to create a new variable.

Reply:
In offline conversation with @agnivade we have agreed to shadow the ResponseWriter variable coming in the `ServerHTTP` method and cast it to the wrapper when needed, for example calling the `StatusCode` method